Fix sass-lint warnings

While testing some CSS changes to the legal pages, I noticed we have a bunch of Sass-lint warnings. I should fix those because they're annoying.

``` sh
✗ gulp sass-lint
[07:45:00] Using gulpfile ~/dev/github/mozilla/testpilot/gulpfile.js
[07:45:00] Starting 'sass-lint'...

testpilot/frontend/static-src/styles/_base.scss
  7:2  warning  Pseudo-elements must start with double colons  pseudo-element
  8:2  warning  Pseudo-elements must start with double colons  pseudo-element

✖ 2 problems (0 errors, 2 warnings)


testpilot/frontend/static-src/styles/_state.scss
  167:40  warning  No trailing whitespace allowed  no-trailing-whitespace

✖ 1 problem (0 errors, 1 warning)


testpilot/frontend/static-src/styles/modules/_avatar.scss
  3:47  warning  No trailing whitespace allowed  no-trailing-whitespace

✖ 1 problem (0 errors, 1 warning)


testpilot/frontend/static-src/styles/modules/_banners.scss
  94:4  warning  No trailing whitespace allowed  no-trailing-whitespace

✖ 1 problem (0 errors, 1 warning)


testpilot/frontend/static-src/styles/modules/_copter.scss
  16:4  warning  No trailing whitespace allowed  no-trailing-whitespace
  39:4  warning  No trailing whitespace allowed  no-trailing-whitespace

✖ 2 problems (0 errors, 2 warnings)


testpilot/frontend/static-src/styles/modules/_experiment-details.scss
  257:4  warning  Pseudo-elements must start with double colons  pseudo-element

✖ 1 problem (0 errors, 1 warning)


testpilot/frontend/static-src/styles/modules/_experiment-list.scss
   63:8   warning  Pseudo-elements must start with double colons  pseudo-element
  115:26  warning  No trailing whitespace allowed                 no-trailing-whitespace
  119:4   warning  Pseudo-elements must start with double colons  pseudo-element

✖ 3 problems (0 errors, 3 warnings)


testpilot/frontend/static-src/styles/modules/_modals.scss
    6:4   warning  Pseudo-elements must start with double colons  pseudo-element
  233:15  warning  No trailing whitespace allowed                 no-trailing-whitespace

✖ 2 problems (0 errors, 2 warnings)


testpilot/frontend/static-src/styles/modules/_settings.scss
  68:4  warning  Pseudo-elements must start with double colons  pseudo-element

✖ 1 problem (0 errors, 1 warning)

[07:45:01] Finished 'sass-lint' after 713 ms
```

---

Ref: https://github.com/sasstools/sass-lint/blob/master/docs/rules/pseudo-element.md

> ### Examples
> 
> When enabled, the following are **allowed**:
> 
> ``` css
> .foo::before {
>   content: "bar";
> }
> .foo:hover {
>   content: "bar";
> }
> ```
> 
> When enabled, the following are **disallowed**:
> 
> ``` css
> .foo:before {
>   content: "bar";
> }
> .foo::hover {
>   content: "bar";
> }
> ```

--- a/testpilot/frontend/static-src/styles/_base.scss
+++ b/testpilot/frontend/static-src/styles/_base.scss
@@ -4,8 +4,8 @@ html {
 }
 
 *,
-*:before,
-*:after {
+*::before,
+*::after {
   box-sizing: inherit;
 }
 

--- a/testpilot/frontend/static-src/styles/_state.scss
+++ b/testpilot/frontend/static-src/styles/_state.scss
@@ -164,7 +164,7 @@
   }
 
   90% {
-    transform: translate3d(0, -5px, 0); 
+    transform: translate3d(0, -5px, 0);
   }
 
   95% {

--- a/testpilot/frontend/static-src/styles/modules/_avatar.scss
+++ b/testpilot/frontend/static-src/styles/modules/_avatar.scss
@@ -1,5 +1,5 @@
 .avatar {
   border-radius: $circle-border-radius;
-  box-shadow: 0 0 0 1px $transparent-black-05, 
+  box-shadow: 0 0 0 1px $transparent-black-05,
   0 0 3px 1px $transparent-black-1;
 }

--- a/testpilot/frontend/static-src/styles/modules/_banners.scss
+++ b/testpilot/frontend/static-src/styles/modules/_banners.scss
@@ -92,7 +92,7 @@
   @include respond-to('not-small') {
     @include flex-container(row-reverse, space-between, center);
   }
-  
+
   @include respond-to('small') {
     @include flex-container(column, center, center);
   }

--- a/testpilot/frontend/static-src/styles/modules/_copter.scss
+++ b/testpilot/frontend/static-src/styles/modules/_copter.scss
@@ -14,7 +14,7 @@
     height: 220px;
     width: 260px;
   }
-  
+
   margin: 0 auto;
   position: relative;
 }
@@ -37,7 +37,7 @@
     height: 220px;
     width: 260px;
   }
-  
+
   background-position: center center;
   background-repeat: no-repeat;
   position: relative;

--- a/testpilot/frontend/static-src/styles/modules/_experiment-details.scss
+++ b/testpilot/frontend/static-src/styles/modules/_experiment-details.scss
@@ -254,8 +254,7 @@
     display: none;
   }
 
-  &:before {
-
+  &::before {
     @media (min-resolution: 1.3dppx) {
       animation: pop 2000ms alternate ease-in-out;
       animation-iteration-count: 5;

--- a/testpilot/frontend/static-src/styles/modules/_experiment-list.scss
+++ b/testpilot/frontend/static-src/styles/modules/_experiment-list.scss
@@ -60,7 +60,7 @@
         box-shadow: 0 0 0 3px $white, 0 0 0 8px $primary;
       }
 
-      &:after {
+      &::after {
         @include hidpi-background-image('check', 36px 27px);
         background-color: $primary;
         background-position: center center;
@@ -112,11 +112,11 @@
   background: $transparent-black-2;
   border-radius: $small-border-radius $small-border-radius 0 0;
   box-shadow: 0 -2px 2px $transparent-white-1 inset;
-  height: $grid-unit * 8; 
+  height: $grid-unit * 8;
   overflow: hidden;
   position: relative;
 
-  &:after {
+  &::after {
     background: $transparent-white-05;
     content: '';
     height: $grid-unit * 15;

--- a/testpilot/frontend/static-src/styles/modules/_modals.scss
+++ b/testpilot/frontend/static-src/styles/modules/_modals.scss
@@ -3,7 +3,7 @@
   margin-bottom: $grid-unit * 1.5;
   position: relative;
 
-  &:after {
+  &::after {
     border-left: $grid-unit * .75 solid transparent;
     border-right: $grid-unit * .75 solid transparent;
     border-top: $grid-unit * .75 solid $white;
@@ -230,7 +230,7 @@
   }
 
   .tour-next,
-  .tour-back { 
+  .tour-back {
     cursor: pointer;
     height: 40px;
     opacity: .7;

--- a/testpilot/frontend/static-src/styles/modules/_settings.scss
+++ b/testpilot/frontend/static-src/styles/modules/_settings.scss
@@ -65,7 +65,7 @@
     text-overflow: ellipsis;
   }
 
-  &:after {
+  &::after {
     border-bottom: $grid-unit * .5 solid $white;
     border-left: ($grid-unit * .5) + 1 solid transparent;
     border-right: ($grid-unit * .5) + 1 solid transparent;

